src/scheduler/agent/database.c

```diff
@@ -215,7 +215,7 @@ static gboolean email_replace(const GMatchInfo* match, GString* ret,
     sql = g_strdup_printf(folder_name, job->id);
     db_result = database_exec(args->scheduler, sql);
 
-    if(PQresultStatus(db_result) != PGRES_TUPLES_OK)
+    if(PQresultStatus(db_result) != PGRES_TUPLES_OK || PQntuples(db_result) == 0)
     {
       g_string_append_printf(ret,
           "[ERROR: unable to select folder name for upload %d]", job->id);
```

Comment:
Would be better if we change `ERROR` to `NOTICE` or `WARNING` as this message will be included in emails sent for maintenance jobs which will never have folders. This can lead to confusion.

```suggestion
          "[NOTICE: Unable to select folder name for upload %d]", job->id);
```